src/main/scala/com/fulcrumgenomics/umi/FilterConsensusReads.scala

```diff
@@ -0,0 +1,217 @@
+      |  2. Bases with fewer than min-reads contributing raw reads are masked to Ns
+      |  3. Bases with a consensus error rate (defined as the fraction of contributing reads that
+      |     voted for a different base than the consensus call) higher than max-base-error-rate
+      |     are masked to Ns
+      |  4. Reads with a proportion of Ns higher than max-no-call-fraction *after* per-base
+      |     filtering are filtered out.
+      |  5. Reads with fewer than min-reads contributing reads are filtered out
+      |  6. Reads with an average consensus error rate higher than max-read-error-rate are
+      |     filtered out
+      |
+      |In order to correctly filter reads in or out by template, if the input BAM is not queryname
+      |sorted or grouped it will be sorted into queryname order.  Likewise, in order to emit
+      |records with corrected NM, UQ and MD tags the reads are re-sorted into coordinate order
+      |for output.
+      |
+      |The reverse-tags-per-base option controls whether per-base tags should be reversed before
+      |being used on reads marked as being mapped to the negative strand.  This is necessary if
+      |the reads have been mapped and the bases/quals reversed but the consensus tags have not.
+      |If true, the tags written to the output BAM will be reversed where necesary in order to
+      |line up with the bases and quals.
+    """
+)
+class FilterConsensusReads
+( @arg(flag="i", doc="The input SAM or BAM file of consensus reads.") val input: PathToBam,
+  @arg(flag="o", doc="Output SAM or BAM file.") val output: PathToBam,
+  @arg(flag="r", doc="Reference fasta file.") val ref: PathToFasta,
+  @arg(flag="R", doc="Reverse [complement] per base tags on reverse strand reads.") val reversePerBaseTags: Boolean = true,
+  @arg(flag="N", doc="Mask (make 'N') consensus bases with quality less than this threshold.") val minBaseQuality: PhredScore = DefaultMinConsensusBaseQuality,
+  @arg(flag="M", doc="The minimum number of reads supporting a consensus base/read.") val minReads: Int = DefaultMinReads,
+  @arg(flag="E", doc="The maximum raw-read error rate across the entire consensus read.") val maxReadErrorRate: Double = 0.025,
+  @arg(flag="e", doc="The maximum error rate for a single consensus base.") val maxBaseErrorRate: Double = 0.1,
+  @arg(flag="n", doc="Maximum fraction of no-calls in the read after filtering.") val maxNoCallFraction: Double = 0.2
+) extends FgBioTool with LazyLogging {
+  // Baseline input validation
+  Io.assertReadable(input)
+  Io.assertReadable(ref)
+  Io.assertCanWriteFile(output)
+  if (maxReadErrorRate < 0 && maxReadErrorRate > 1) fail("max-read-error-rate must be between 0 and 1.")
+  if (maxBaseErrorRate < 0 && maxBaseErrorRate > 1) fail("max-base-error-rate must be between 0 and 1.")
+  if (maxNoCallFraction < 0 && maxNoCallFraction > 1) fail("max-no-call-fraction must be between 0 and 1.")
```

Comment:
What about a `failif` method that takes in a boolean and a message, similar to `require`?